moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -317,6 +339,12 @@ bool PlanningPipeline::generatePlan(const planning_scene::PlanningSceneConstPtr&
       context->solve(res);
       publishPipelineState(mutable_request, res, planner->getDescription());
 
+      // check for overall timeout
+      if (std::chrono::duration<double>(clock::now() - plan_start_time).count() >= allowed_planning_time)
```

Comment:
Maybe I'm missing it, but it looks like we will hit a timeout here if the planners use up the allowed planning time. Optimizing planners usually do that, so I think we should either ignore response adapters or use a separate budget